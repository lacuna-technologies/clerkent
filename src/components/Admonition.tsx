import { useCallback, useState } from 'preact/hooks'
import Helpers from '../utils/Helpers'

const Admonition = ({
  className = ``,
  title = ``,
  children = null,
  onClick = null,
}) => {
  const [shouldHide, setShouldHide] = useState(false)
  const dismissAdmonition = useCallback((event) => {
    event.stopPropagation()
    setShouldHide(true)
  }, [])

  return shouldHide ? null : (
    <div
      className={Helpers.classnames(
        `border border-solid border-gray-400 px-2 py-1 flex flex-row hover:bg-gray-100`,
        className,
        onClick ? `cursor-pointer hover:bg-100` : ``,
      )}
      onClick={onClick ? onClick : () => {}}
    >
      <div className="grow">
        <div className="inline-flex flex-row">
          <strong>{title}</strong>
        </div>
        <div>
          {children}
        </div>
      </div>
      <div className="cursor-pointer text-xs p-1" onClick={dismissAdmonition}>âœ•</div>
    </div>
  )
}

export default Admonition