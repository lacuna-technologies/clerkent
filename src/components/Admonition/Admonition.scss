@import "../../styles/core.scss";

.admonition {
  border: 1px solid $border-color;
  padding: 0.5rem;
  display: flex;
  flex-direction: row;
  box-sizing: border-box;

  &.clickable {
    cursor: pointer;
    &:hover {
      background-color: darken(white, 5%);
    }
  }

  > .left {
    flex-grow: 1;
    > .header {
      display: inline-flex;
      flex-direction: row;
    }
  }

  > .right {
    > .dismiss-button {
      cursor: pointer;
      font-size: 75%;
    }
  }  
}