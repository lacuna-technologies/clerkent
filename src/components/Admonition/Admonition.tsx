import React, { useCallback, useState } from 'react'
import Helpers from '../../utils/Helpers'
import './Admonition.scss'

const Admonition = ({
  className = ``,
  title = ``,
  children = null,
  onClick = null,
}) => {
  const [shouldHide, setShouldHide] = useState(false)
  const dismissAdmonition = useCallback((event) => {
    event.stopPropagation()
    setShouldHide(true)
  }, [])

  return shouldHide ? null : (
    <div
      className={Helpers.classnames(`admonition`, className, onClick ? `clickable` : ``)}
      onClick={onClick ? onClick : () => {}}
    >
      <div className="left">
        <div className="header">
          <strong>{title}</strong>
        </div>
        <div className="content">
          {children}
        </div>
      </div>
      <div className="right">
        <div className="dismiss-button" onClick={dismissAdmonition}>âœ•</div>
      </div>
    </div>
  )
}

export default Admonition