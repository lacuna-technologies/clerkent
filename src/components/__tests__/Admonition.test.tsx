import Admonition from "components/Admonition"
import { render } from '@testing-library/preact'

describe(`Admonition`, () => {
  it(`should render without errors`, () => {
    const tree = render(
      <Admonition title="Warning">
        <p>
          Fun will now commence
        </p>
      </Admonition>,
    )
    expect(tree).toMatchSnapshot()
  })
})