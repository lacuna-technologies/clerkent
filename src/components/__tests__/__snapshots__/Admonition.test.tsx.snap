// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admonition should render without errors 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="border border-solid border-gray-400 px-2 py-1 flex flex-row hover:bg-gray-100 cursor-pointer hover:bg-100"
      >
        <div
          class="grow"
        >
          <div
            class="inline-flex flex-row"
          >
            <strong>
              Warning
            </strong>
          </div>
          <div>
            <p>
              Fun will now commence
            </p>
          </div>
        </div>
        <div
          class="cursor-pointer text-xs p-1"
        >
          ✕
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="border border-solid border-gray-400 px-2 py-1 flex flex-row hover:bg-gray-100 cursor-pointer hover:bg-100"
    >
      <div
        class="grow"
      >
        <div
          class="inline-flex flex-row"
        >
          <strong>
            Warning
          </strong>
        </div>
        <div>
          <p>
            Fun will now commence
          </p>
        </div>
      </div>
      <div
        class="cursor-pointer text-xs p-1"
      >
        ✕
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
